Sources/Views/MessageInputBar.swift

```diff
@@ -251,26 +251,52 @@ open class MessageInputBar: UIView {
             ).activate()
         textViewHeightAnchor = inputTextView.heightAnchor.constraint(equalToConstant: maxHeight)
         
-        leftStackViewLayoutSet = NSLayoutConstraintSet(
-            top:    inputTextView.topAnchor.constraint(equalTo: topAnchor, constant: padding.top),
-            bottom: leftStackView.bottomAnchor.constraint(equalTo: inputTextView.bottomAnchor, constant: 0),
-            left:   leftStackView.leftAnchor.constraint(equalTo: leftAnchor, constant: padding.left),
-            width:  leftStackView.widthAnchor.constraint(equalToConstant: leftStackViewWidthContant)
+        if #available(iOS 11.0, *) {
+            
+            leftStackViewLayoutSet = NSLayoutConstraintSet(
+                top:    inputTextView.topAnchor.constraint(equalTo: topAnchor, constant: padding.top),
+                bottom: leftStackView.bottomAnchor.constraint(equalTo: inputTextView.bottomAnchor, constant: 0),
+                left:   leftStackView.leftAnchor.constraint(equalTo: safeAreaLayoutGuide.leftAnchor, constant: padding.left),
+                width:  leftStackView.widthAnchor.constraint(equalToConstant: leftStackViewWidthContant)
             ).activate()
-        
-        rightStackViewLayoutSet = NSLayoutConstraintSet(
-            top:    inputTextView.topAnchor.constraint(equalTo: topAnchor, constant: padding.top),
-            bottom: rightStackView.bottomAnchor.constraint(equalTo: inputTextView.bottomAnchor, constant: 0),
-            right:  rightStackView.rightAnchor.constraint(equalTo: rightAnchor, constant: -padding.right),
-            width:  rightStackView.widthAnchor.constraint(equalToConstant: rightStackViewWidthContant)
+            
+            rightStackViewLayoutSet = NSLayoutConstraintSet(
+                top:    inputTextView.topAnchor.constraint(equalTo: topAnchor, constant: padding.top),
+                bottom: rightStackView.bottomAnchor.constraint(equalTo: inputTextView.bottomAnchor, constant: 0),
+                right:  rightStackView.rightAnchor.constraint(equalTo: safeAreaLayoutGuide.rightAnchor, constant: -padding.right),
+                width:  rightStackView.widthAnchor.constraint(equalToConstant: rightStackViewWidthContant)
             ).activate()
-        
-        bottomStackViewLayoutSet = NSLayoutConstraintSet(
-            top:    bottomStackView.topAnchor.constraint(equalTo: inputTextView.bottomAnchor),
-            bottom: bottomStackView.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor, constant: -padding.bottom),
-            left:   bottomStackView.leftAnchor.constraint(equalTo: leftAnchor, constant: padding.left),
-            right:  bottomStackView.rightAnchor.constraint(equalTo: rightAnchor, constant: -padding.right)
```

Comment:
Looks like we have some repeated constraints here, maybe we can extract them for reuse in the initializer?

Reply:
good idea